src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -161,17 +178,41 @@ class Manager : public DoutPrefixProvider {
     }   
   };
 
+  enum class EntryProcessingResult {
+    Failure, Successful, Sleeping, Expired
+  };
+
   // processing of a specific entry
   // return whether processing was successfull (true) or not (false)
-  bool process_entry(const cls_queue_entry& entry, yield_context yield) {
+  EntryProcessingResult process_entry(const Manager& manager, persistency_tracker& entry_persistency_tracker,
+                                      const cls_queue_entry& entry, yield_context yield) {
     event_entry_t event_entry;
     auto iter = entry.data.cbegin();
     try {
       decode(event_entry, iter);
     } catch (buffer::error& err) {
       ldpp_dout(this, 5) << "WARNING: failed to decode entry. error: " << err.what() << dendl;
-      return false;
+      return EntryProcessingResult::Failure;
+    }
+
+    const auto conf = manager.get_cct()->_conf;
+    if ( (event_entry.creation_time != ceph::coarse_real_clock::zero() &&
+          ceph::coarse_real_clock::now() - event_entry.creation_time > std::chrono::seconds(manager.get_cct()->_conf->rgw_topic_persistency_time_to_live))
```

Comment:
please cache `ceph::coarse_real_clock::now()` to a local var instead of calling it twice